lib/libxdp/libxdp.c

```diff
@@ -2112,40 +2169,61 @@ static int xdp_multiprog__attach(struct xdp_multiprog *old_mp,
 				 struct xdp_multiprog *mp,
 				 enum xdp_attach_mode mode)
 {
-	int err = 0, prog_fd = -1, old_fd = -1, ifindex = -1;
+	int err = 0, prog_fd = -1, old_fd = -1, hw_fd = -1, ifindex = -1;
```

Comment:
I think it's better to leave HW mode out of this function entirely, and just create a separate function to handle that...